framework/src/org/checkerframework/framework/test/ImmutableTestConfiguration.java

```diff
@@ -0,0 +1,87 @@
+package org.checkerframework.framework.test;
+
+import org.checkerframework.framework.util.PluginUtil;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Represents all of the information needed to execute the Javac compiler for a given set of test files.
+ */
+public class ImmutableTestConfiguration implements TestConfiguration {
+
+    /**
+     * Options that should be passed to the compiler
```

Comment:
I take that back.  The reason this is a map is because of the above reasoning but the ImmutableTestConfiguration should not contain the SimpleOptionMap (because it is mutable).